client/state/lasagna/post-channel/middleware.js

```diff
@@ -57,14 +57,19 @@ const joinChannel = async ( store, joinParams ) => {
 			return;
 		}
 
-		store.dispatch(
-			receiveComments( {
-				siteId: comment.post.site_ID,
-				postId: comment.post.ID,
-				comments: [ comment ],
-				commentById: true,
-			} )
-		);
+		// If the comment's author is 0, that means it was authored by the current user.
+		// In that case do not dispatch the event as it may arrive before the pending comment has been deleted,
+		// causing a duplicate comment. See https://github.com/Automattic/wp-calypso/issues/61812
```

Comment:
Thanks for adding an explanation 👍

Reply:
yes this was a confusing one but I think it makes sense with an explanation. Thank you 😄